src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains the ticking function that is responsible for invoking block acquisition functionality and executing it in periodic fashion.
+
+
+module Cardano.Wallet.BlockSyncer
+  (
+    blockSyncerBracket
+  ) where
+
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Block (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Exception
+    ( SomeException, bracket, catch )
+import Data.Text.Internal
+    ( showText )
+import Formatting
+    ( sformat, (%) )
+
+import qualified Formatting as F
+
+
+blockSyncerBracket
+    :: IO [Block]
+    -- ^ a way to get a new block
+    -> (Block -> IO ())
+    -- ^ action taken on a new block
+    -> Int
+    -- ^ tick time in seconds
+    -> IO ()
+blockSyncerBracket getNextBlocks action tickTime =
+    bracket
+      (return ())
+      (\_->return ())
+      (tickingFunction getNextBlocks action tickTime)
+
+tickingFunction
+    :: IO [Block]
+    -> (Block -> IO ())
+    -> Int
```

Comment:
Would be better to get a better type here which convey a clear meaning -- or... not having any at all. The time can just be a hard-coded constant on top of the module, if we ever need to change that, we'll make it a parameter. But again, for now, this parameter isn't needed or required.

Reply:
went with Second from `time-units`